### DISCUSSION TITLE
boolean brush

### DISCUSSION THREAD

**Discussion by Ulf3000** [chsh2/nijiGPen#35] [Ideas]
hey man i often use your boolean method now. thanx to it i can create more advanced shapes easier. would be awesome if there'd be an auto boolean for the brush. i have it on a shortcut which is also fast but maybe a missed opportunity.

**Comment by chsh2** [comment]
I would like to make it automatic, but I do not know how. I think it requires us to trigger the boolean function once a new stroke is drawn. However, I cannot find a [handler](https://docs.blender.org/api/current/bpy.app.handlers.html) for this event. Maybe it is not allowed by Blender.

Do you have ideas about how to implement it? Or have you seen add-ons with similar functionality, i.e., executing some operations automatically after using a brush?

**Comment by chsh2** [comment]
I am reevaluating the possibility of implementing this. During developing the interactive mode of Smart Fill, I got some experience about how to create a new 3D View tool in Blender. Maybe I can do Boolean in the same way, by creating a new tool, which captures user's mouse/stylus movements.

**Reply by ianGchang** [reply]
Hey, I love using this add-on. Having a "brush" that does the boolean operation automatically after a stroke would be amazing and bring it to the level of other art software with their "Laso Fill/Erase" tools. I'm currently using it with the Quick Menu, which requires a couple extra steps.
Do you have any news on possibly adding this option?
Thank you!

**Reply by chsh2** [reply]
I made a [boolean eraser](https://chsh2.github.io/nijigp/docs/functionality/polygon/bool/#boolean-eraser-tool) tool to perform boolean operations automatically. Currently, it behaves like an eraser, but maybe I can also add a lasso mode in the next version.

One shortcoming of this tool is that it is less fluent than the native brushes. There are also fewer smoothing options supported.

**Reply by ianGchang** [reply]
Thank you, I wasn't aware of it. It doesn't seem to be as reliable as the the Polygon operators (+, -), most of the time it just separate the shapes for me, and rarely it'll delete them. Am I doing something wrong?

Video comparison between nijiGPen's Boleean eraser and Polygon Operators
https://github.com/user-attachments/assets/4f95fc2c-255c-4e9c-a7cf-0b3d335d3aac

The laso fill (to add shapes) tool is already there with Blender's base grease pencil functionallity, unless we understand "Laso fill" differently. Is just the "eraser" mode that's missing for me in base Blender and that's why I found nijiGPen so valuable. As you can see in my video, using nijiGpen's Polygon operator through the Quick menu is pretty quick already, it's just that the Boolean eraser (Laso fill erase) idea is better suited for my needs (saves you one click/shortcut)

Again, thanks for the amazing add-on.

**Reply by chsh2** [reply]
https://github.com/user-attachments/assets/6921c881-44dd-49a2-b316-0904e27155ec

Currently, the tool requires the user to smear over the area to erase. In the next version, I will add a new option in this tool to perform the operation you described.

I had to create this new tool because Blender does not seem to allow add-ons to change the way its base tools and brushes work. Although it is only one click/shortcut, I cannot find a way to execute it automatically. So I chose this more complicated implementation.

**Reply by ianGchang** [reply]
Ooh, now I'm clear on how the current mode is supposed to work. The new mode looks exactly what I was looking for, can't wait to eventually try it. Thank you!
